bin/run-utils.js

```diff
@@ -184,3 +184,195 @@ function runFirefox(options) {
+    return false;
+  }
+  return value;
+}
+
+function takePrefsSample(profilePath) {
+  currentSample = [];
+  currentSampleMap = {};
+
+  return When.promise(function(resolve, reject) {
+    var prefsFile = Path.join(profilePath, "user.js");
+    Fs.exists(prefsFile).then(function(exists) {
+      if (!exists) {
+        resolve();
+        return;
+      }
+
+      Fs.readFile(prefsFile, "utf8").then(function(content) {
+        currentSample = content.split("\n").map(function(line) {
+          var found = line.match(prefsRE);
+          if (found) {
+            var prefName = found[1];
+            var prefValue = parsePrefValue(found[2]);
+
+            // Check for JSON values:
+            var parts = prefName.split(/['"],\s*['"]/);
+            if (parts.length > 1) {
+              prefName = parts.shift();
+              prefValue = parts.join("\",\"") + prefValue;
+            }
+
+            currentSampleMap[prefName] = prefValue;
+            // Sample the name only, to be able to reference it later when we
+            // write the preferences back to file.
+            return prefName;
+          }
+          return line;
+        });
+
+        // Clear trailing empty lines.
+        var lastLine = currentSample.pop();
+        while (!lastLine.replace(/^[\s\n\t\r]+$/, "")) {
+          lastLine = currentSample.pop();
+        }
+        if (lastLine) {
+          currentSample.push(lastLine);
+        }
+
+        resolve();
+      }).catch(reject);
+    });
+  });
+}
+
+function prefValueToString(val) {
+  switch (typeof val) {
+    case "boolean":
+      return val ? "true" : "false";
+    case "number":
+      return "" + val;
+    case "string":
+      return "\"" + val.replace(/[^\\]{0}"/g, "\\\"") + "\"";
+    default:
+      return val;
+  }
+}
+
+function getUserPrefLine(name, value) {
+  return "user_pref(\"" + name + "\", " + prefValueToString(value) + ");";
+}
+
+function writeUserPrefs(profilePath, userPrefs) {
+  if (!userPrefs) {
+    userPrefs = currentSample || [];
+  }
+
+  var prefsFile = Path.join(profilePath, "user.js");
+  return Fs.writeFile(prefsFile, userPrefs.join("\n") + "\n", "utf8");
+}
+
+function setUserPrefs(profilePath, prefs) {
+  return takePrefsSample(profilePath)
+    .then(function() {
+      var seen = {};
+      var userPrefs = currentSample.map(function(entry) {
+        if (entry in currentSampleMap) {
+          // We encountered a line containing a pref.
+          var prefValue = currentSampleMap[entry];
+          if (entry in prefs) {
+            // This line of the prefs needs to be overwritten.
+            prefValue = prefs[entry];
+            seen[entry] = 1;
+          }
+
+          return getUserPrefLine(entry, prefValue);
+        }
+
+        return entry;
+      });
+
+      Object.getOwnPropertyNames(prefs).forEach(function(entry) {
+        if (seen[entry]) {
+          // An existing pref line has been overwritten above.
+          return;
+        }
```

Comment:
With all the above, it's nice to be clean and correctly replace existing prefs with the defaults, but `prefs.js` file works just fine with duplicate `user_pref()` calls with the same first argument. The later calls overwrite the earlier ones. This means we can just append to the file the default prefs we want.